### PR TITLE
AEA-809 Update GitHub Actions workflow

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Update GitHub Actions workflow.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -121,7 +119,7 @@ jobs:
     continue-on-error: True
     runs-on: ubuntu-latest
 
-    timeout-minutes: 40
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yml:124]: This one should only be 30 minutes

File: tox.ini [hunk 2]
```diff
@@ -36,7 +36,7 @@ deps =
     pytest-rerunfailures==9.0
 
 commands = pip install -i https://test.pypi.org/simple/ fetch-p2p-api==0.0.2
-           pytest -rfE --doctest-modules aea packages/fetchai/protocols packages/fetchai/connections tests/ --cov-report=html --cov-report=xml --cov-report=term --cov=aea --cov=packages/fetchai/protocols --cov=packages/fetchai/connections {posargs}
```
Comment (DavidMinarsch) [inline tox.ini:39]: We don't yet have full coverage for protocols in packages; so excluding them for now

Comment (marcofavorito) [reply tox.ini:39]: ok

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -110,48 +46,104 @@ jobs:
...
+      with:
+        go-version: '^1.14.0'
+    - name: Install dependencies (ubuntu-latest)
+      run: |
+        sudo apt-get update --fix-missing
+        sudo apt-get autoremove
+        sudo apt-get autoclean
+        pip install tox
+        sudo apt-get install -y protobuf-compiler
+    - name: Sync AEA loop integrational tests and coverage
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:106]: ... ` and coverage`? Shouldn't this be removed?

File: .github/workflows/workflow.yml [hunk 4]
```diff
@@ -110,48 +46,104 @@ jobs:
...
+
+  common_checks_2:
+    continue-on-error: False
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-python@master
+        with:
+          python-version: 3.6
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:58]: why Python `3.6`? And not e.g. `3.8`?

File: .github/workflows/workflow.yml [hunk 5]
```diff
@@ -8,63 +8,11 @@ on:
...
-        pip install pipenv
-        pip install tox
-        sudo apt-get install -y protobuf-compiler
-    - name: Unit tests and coverage
-      run: |
-        tox -e py3.8 -- --aea-loop sync -m 'not integration and not unstable'
 
-  all_integrational_tests:
-    continue-on-error: True
+  common_checks_1:
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:12]: [minor comment] I would put a comment here to describe what's `common_checks_1` about (I presume you meant Python linting). Also, I would consider changing the name of the job to `python_linting`, or just `linting`, or similar.

File: .github/workflows/workflow.yml [hunk 6]
```diff
@@ -166,50 +158,63 @@ jobs:
...
         brew install protobuf
     - if: matrix.os == 'windows-latest'
       name: Install dependencies (windows-latest)
       run: |
-        pip install pipenv
         pip install tox
         echo "::add-path::C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x64"
         choco install protoc
         python scripts/update_symlinks_cross_platform.py
     - name: Unit tests and coverage
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:177]: see above re. coverage

File: .github/workflows/workflow.yml [hunk 7]
```diff
@@ -166,50 +158,63 @@ jobs:
...
+      with:
+        go-version: '^1.14.0'
+    - name: Install dependencies (ubuntu-latest)
+      run: |
+        sudo apt-get update --fix-missing
+        sudo apt-get autoremove
+        sudo apt-get autoclean
+        pip install tox
+        sudo apt-get install -y protobuf-compiler
+    - name: Unit tests and coverage
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:203]: see above re. coverage

File: .github/workflows/workflow.yml [hunk 8]
```diff
@@ -166,50 +176,65 @@ jobs:
...
+      run: |
+        tox -e py3.8 -- --aea-loop sync -m 'not integration and not unstable'
+
   golang_checks:
+    continue-on-error: false
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:227]: was it wanted? The other jobs have `true`

File: .github/workflows/workflow.yml (hunk 4 above)
Comment (DavidMinarsch) [reply .github/workflows/workflow.yml:91]: I think for the common check it makes sense to take the lowest version?

File: .github/workflows/workflow.yml (hunk 5 above)
Comment (DavidMinarsch) [reply .github/workflows/workflow.yml:12]: Its obvious from looking at the code/yaml spec; comments then always end up out of sync (see above ;))

File: .github/workflows/workflow.yml (hunk 8 above)
Comment (DavidMinarsch) [reply .github/workflows/workflow.yml:249]: I will change

File: .github/workflows/workflow.yml (hunk 5 above)
Comment (marcofavorito) [reply .github/workflows/workflow.yml:12]: ok

File: .github/workflows/workflow.yml (hunk 4 above)
Comment (marcofavorito) [reply .github/workflows/workflow.yml:91]: ok

**Pull request merged**